### ISSUE TITLE
staff styles/alternative notation

### ISSUE THREAD

**Issue by sannedubois** [rism-ch/verovio#882]
It would be nice to have more possibilities to display other forms of notation for chords. In the next example there is used 1)slash notation, 2) rhythmic notation, 3) a 'cutaway' for a coda, 4) one bar repeats and 2 bar repeats:
![staffstyles](https://user-images.githubusercontent.com/16167955/40768596-1a0f4cc6-64b6-11e8-8d24-ccc937a6262a.jpg)
Verovio imports the music-xml. We see an interpretation of a one (?) bar repeat, but I do not really know what's meant by it.
![staff style musicxml import](https://user-images.githubusercontent.com/16167955/40768649-3ea509fe-64b6-11e8-9717-f9d26580fb68.png)
The music xml looks like this:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise version="3.0">
  <identification>
    <encoding>
      <software>Finale 2012 for Mac</software>
      <software>Dolet Light for Finale 2012</software>
      <encoding-date>2018-05-31</encoding-date>
      <supports attribute="new-system" element="print" type="yes" value="yes"/>
      <supports attribute="new-page" element="print" type="yes" value="yes"/>
    </encoding>
  </identification>
  <defaults>
    <scaling>
      <millimeters>7.2319</millimeters>
      <tenths>40</tenths>
    </scaling>
    <page-layout>
      <page-height>1545</page-height>
      <page-width>1194</page-width>
      <page-margins type="both">
        <left-margin>70</left-margin>
        <right-margin>70</right-margin>
        <top-margin>88</top-margin>
        <bottom-margin>88</bottom-margin>
      </page-margins>
    </page-layout>
    <system-layout>
      <system-margins>
        <left-margin>0</left-margin>
        <right-margin>0</right-margin>
      </system-margins>
      <system-distance>121</system-distance>
      <top-system-distance>70</top-system-distance>
    </system-layout>
    <appearance>
      <line-width type="stem">0.7487</line-width>
      <line-width type="beam">5</line-width>
      <line-width type="staff">0.7487</line-width>
      <line-width type="light barline">0.7487</line-width>
      <line-width type="heavy barline">5</line-width>
      <line-width type="leger">0.7487</line-width>
      <line-width type="ending">0.7487</line-width>
      <line-width type="wedge">0.7487</line-width>
      <line-width type="enclosure">0.7487</line-width>
      <line-width type="tuplet bracket">0.7487</line-width>
      <note-size type="grace">60</note-size>
      <note-size type="cue">60</note-size>
      <distance type="hyphen">120</distance>
      <distance type="beam">8</distance>
    </appearance>
    <music-font font-family="Maestro,engraved" font-size="20.5"/>
    <word-font font-family="Times New Roman" font-size="10.25"/>
  </defaults>
  <credit page="1">
    <credit-words default-x="70" default-y="1453" font-size="12" valign="top">Score</credit-words>
  </credit>
  <part-list>
    <score-part id="P1">
      <part-name print-object="no">MusicXML Part</part-name>
      <score-instrument id="P1-I1">
        <instrument-name>Grand Piano</instrument-name>
      </score-instrument>
      <midi-instrument id="P1-I1">
        <midi-channel>1</midi-channel>
        <midi-program>1</midi-program>
        <volume>80</volume>
        <pan>0</pan>
      </midi-instrument>
    </score-part>
  </part-list>
  <!--=========================================================-->
  <part id="P1">
    <measure number="1" width="182">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>70</left-margin>
            <right-margin>0</right-margin>
          </system-margins>
          <top-system-distance>211</top-system-distance>
        </system-layout>
        <measure-numbering>system</measure-numbering>
      </print>
      <attributes>
        <divisions>2</divisions>
        <key>
          <fifths>0</fifths>
          <mode>major</mode>
        </key>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
        </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
        </clef>
        <measure-style>
          <slash type="start" use-stems="no"/>
        </measure-style>
      </attributes>
      <sound tempo="120"/>
      <harmony default-y="40">
        <root>
          <root-step>C</root-step>
        </root>
        <kind halign="center" text="">major</kind>
      </harmony>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="2" width="267">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="3" width="267">
      <harmony default-y="40">
        <root>
          <root-step>G</root-step>
        </root>
        <kind halign="center" text="7">dominant</kind>
      </harmony>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="4" width="267">
      <harmony default-y="40">
        <root>
          <root-step>C</root-step>
        </root>
        <kind halign="center" text="">major</kind>
      </harmony>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="5" width="292">
      <print new-system="yes">
        <system-layout>
          <system-distance>114</system-distance>
        </system-layout>
      </print>
      <attributes>
        <measure-style>
          <slash type="stop"/>
        </measure-style>
        <measure-style>
          <slash type="start" use-stems="yes"/>
        </measure-style>
      </attributes>
      <note default-x="52">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem default-y="-65.5">down</stem>
      </note>
      <note default-x="105">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">begin</beam>
      </note>
      <note default-x="139">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">end</beam>
      </note>
      <note default-x="172">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem default-y="-65.5">down</stem>
      </note>
      <note default-x="225">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">begin</beam>
      </note>
      <note default-x="258">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">end</beam>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="6" width="254">
      <note default-x="13">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem default-y="-65.5">down</stem>
      </note>
      <note default-x="67">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">begin</beam>
      </note>
      <note default-x="100">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">end</beam>
      </note>
      <note default-x="133">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem default-y="-65.5">down</stem>
      </note>
      <direction placement="above">
        <direction-type>
          <words default-y="19" font-family="Times">to Coda</words>
        </direction-type>
      </direction>
      <note default-x="187">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">begin</beam>
      </note>
      <note default-x="220">
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem default-y="-65">down</stem>
        <beam number="1">end</beam>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="7" width="254">
      <attributes>
        <measure-style>
          <slash type="stop"/>
        </measure-style>
        <measure-style>
          <measure-repeat type="start">1</measure-repeat>
        </measure-style>
      </attributes>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem>down</stem>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">begin</beam>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">end</beam>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem>down</stem>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">begin</beam>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">end</beam>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="8" width="254">
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem>down</stem>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">begin</beam>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">end</beam>
      </note>
      <direction placement="above">
        <direction-type>
          <words default-y="19" font-family="Times" relative-x="15">D.C. al Coda</words>
        </direction-type>
      </direction>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>2</duration>
        <voice>1</voice>
        <type>quarter</type>
        <stem>down</stem>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">begin</beam>
      </note>
      <note>
        <pitch>
          <step>C</step>
          <octave>5</octave>
        </pitch>
        <duration>1</duration>
        <voice>1</voice>
        <type>eighth</type>
        <stem>down</stem>
        <beam number="1">end</beam>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="9" width="292">
      <print new-system="yes">
        <system-layout>
          <system-distance>114</system-distance>
        </system-layout>
      </print>
      <attributes>
        <measure-style>
          <measure-repeat type="stop"/>
        </measure-style>
        <measure-style>
          <measure-repeat slashes="2" type="start">2</measure-repeat>
        </measure-style>
      </attributes>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="10" width="254">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="11" width="254">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="12" width="254">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="13" width="262">
      <print new-system="yes">
        <system-layout>
          <system-distance>114</system-distance>
        </system-layout>
      </print>
      <attributes>
        <staff-details print-object="no" print-spacing="yes"/>
        <measure-style>
          <measure-repeat type="stop"/>
        </measure-style>
        <measure-style>
          <slash type="start" use-stems="no"/>
        </measure-style>
      </attributes>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="14" width="267">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
      <barline location="right">
        <bar-style>light-light</bar-style>
      </barline>
    </measure>
    <!--=======================================================-->
    <measure number="15" width="261">
      <attributes>
        <staff-details print-object="yes"/>
      </attributes>
      <direction placement="above">
        <direction-type>
          <words default-y="24" font-family="Times" relative-x="6">Coda</words>
        </direction-type>
      </direction>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="16" width="263">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
      </barline>
    </measure>
  </part>
  <!--=========================================================-->
</score-partwise>
```

**Comment by rettinghaus** [comment]
Could you provide a sample, how this should be encoded in MEI?

**Comment by sannedubois** [comment]
slash notation is in fact a beat repeat, so it could look like this (in 4/4):
![slash notation](https://user-images.githubusercontent.com/16167955/40906707-e2216f54-67e1-11e8-914e-8463cfdced4c.jpg)


```<staff>
      <layer>
         <beatRpt></beatRpt>
         <beatRpt></beatRpt>
         <beatRpt></beatRpt>
         <beatRpt></beatRpt>
      </layer>
   </staff>```

**Comment by sannedubois** [comment]
According to http://music-encoding.org/guidelines the repetition of measures should be like this:

![1 bar repeat](https://user-images.githubusercontent.com/16167955/40907318-bfc3b654-67e3-11e8-8de5-6cc7ca5db951.jpg)

```
<measure>
   <staff>
      <layer>
         <mRpt></mRpt>
      </layer>
   </staff>
</measure>
```
A 2-bar repeat like this:

![2 bar repeat](https://user-images.githubusercontent.com/16167955/40907399-ff4d4dd0-67e3-11e8-9c8b-79d760d1db0a.jpg). The number in 'mRpt2' indicated the number of measures repeated. Maybe that would be coded like this?
```
<measure>
   <staff>
      <layer>
         <mRpt2>
      </layer>
   </staff>
</measure>
<measure>
   <staff>
      <layer>
         </mRpt2>
      </layer>
   </staff>
</measure>
```
for 3 measures that would be probably this, because I see that only 3 barlines are used:
```
<measure>
   <staff>
      <layer>
         <mRpt3>
      </layer>
   </staff>
</measure>
<measure>
   <staff>
      <layer>
         </mRpt2>
      </layer>
   </staff>
</measure>
```
If there is text above the multimeasure rests (especially when this is about D.C. al Code etc.) the multimeasure rests/repeats should be interpreted in a way, that they are still there after music xml is imported. In my first example, the multimeasure repeats are not interpreted correctly.

**Comment by sannedubois** [comment]
The 'cutaway' (as Finale 2012) names it looks like this, and is used often before a coda appears:

![break](https://user-images.githubusercontent.com/16167955/40908219-645cb5d8-67e6-11e8-8b97-9e65ee5e1aae.jpg)
music-xml shows the cutaway like this:
```
<measure number="13" width="262">
      <print new-system="yes">
        <system-layout>
          <system-distance>114</system-distance>
        </system-layout>
      </print>
      <attributes>
        <staff-details print-object="no" print-spacing="yes"/>
        <measure-style>
          <measure-repeat type="stop"/>
        </measure-style>
        <measure-style>
          <slash type="start" use-stems="no"/>
        </measure-style>
      </attributes>
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
    </measure>
    <!--=======================================================-->
    <measure number="14" width="267">
      <note>
        <rest measure="yes"/>
        <duration>8</duration>
        <voice>1</voice>
      </note>
      <barline location="right">
        <bar-style>light-light</bar-style>
      </barline>
    </measure>
    <!--=======================================================-->
    <measure number="15" width="261">
      <attributes>
        <staff-details print-object="yes"/>
      </attributes>
```
In MEI I see only a <sb> tag for the  first measure (no.13) that is hidden:

```                        <sb xml:id="sb-0000000634866949" />
                        <measure xml:id="measure-0000000764994411" n="13">
                            <staff xml:id="staff-0000000820458311" n="1">
                                <layer xml:id="layer-0000000815439006" n="1">
                                    <mRest xml:id="mrest-0000001653432866" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-0000001946492142" right="dbl" n="14">
                            <staff xml:id="staff-0000002034234354" n="1">
                                <layer xml:id="layer-0000000316318054" n="1">
                                    <mRest xml:id="mrest-0000002054402477" />
                                </layer>
                            </staff>
```
I could not find if MEI actually can support this.

**Comment by rettinghaus** [comment]
Slash notation and cutaways implemented with #887. 
However your cutaways will not work properly, as the Sibelius implementation is wrong, as you can see in the MusicXML Documentation: 
https://usermanuals.musicxml.com/MusicXML/Content/CT-MusicXML-staff-details.htm

**Comment by lpugin** [comment]
Can we close it?
